src/pages/projectSettings/tabs/GeneralTab/GeneralTab.tsx

```diff
@@ -67,6 +67,27 @@ export const GeneralTab: React.VFC<TabProps> = ({
       onChange={onChange}
       schema={schema}
       uiSchema={uiSchema}
+      validate={validate(projectType) as any}
     />
   );
 };
+
+const validate = (projectType: ProjectType) =>
+  ((formData, errors) => {
+    const {
+      generalConfiguration: { enabled, branch },
+    } = formData;
+
+    // Repos don't use the enabled field and are always considered enabled, so don't check for it.
+    const invalidRepoBranch = projectType === ProjectType.Repo && !branch;
+    const invalidProjectBranch =
+      projectType === ProjectType.Project && enabled && !branch;
+
+    if (invalidRepoBranch || invalidProjectBranch) {
+      errors.generalConfiguration.branch.addError(
+        "A branch is required for enabled projects."
+      );
+    }
+
```

Comment:
maybe this is an unrealistic scenario, but I can get around the requirement by doing this:

https://user-images.githubusercontent.com/47064971/235202896-37cebaaf-b6b1-4980-b5be-142b0fac43eb.mov

Reply:
I think this is okay, if the user is attaching to a repo for the first time it's likely they'll look into the repo settings where this issue is clearly flagged. But thank you for raising!